src/components/MultiHopTrade/hooks/useGetTradeQuotes/getTradeQuoteInput.ts

```diff
@@ -34,14 +25,30 @@ export type GetTradeQuoteInputArgs = {
   affiliateBps: string
   isSnapInstalled?: boolean
   pubKey?: string | undefined
-}
+} & (
+  | {
+      receiveAccountNumber?: number
+      receiveAddress: string
+      sellAccountNumber: number
+      wallet: HDWallet
+      hasWallet: true
+    }
+  | {
+      receiveAccountNumber?: number
+      receiveAddress: string | undefined
+      sellAccountNumber: number | undefined
+      wallet: HDWallet | undefined
+      hasWallet: false
+    }
+)
```

Comment:
this union doesn't seem important at this level imo. we are just passing through the args either defined or not and the swappers handle the discrimination accordingly.

Reply:
@kaladinlight don't we feel like it's better to have two clearly separate types, regardless of the discrimination being handled at runtime or not?

Reply:
Actually reading https://github.com/shapeshift/web/pull/7939#discussion_r1809643839, this does make sense in terms of ergonomics, taking a look now

Reply:
6f13836f3e